lib/view_component/collection.rb

```diff
@@ -69,5 +70,15 @@ def component_options(item, iterator)
 
       @options.merge(item_options)
     end
+
+    def rendered_spacer(view_context)
+      if @spacer_component
+        spacer = @spacer_component.new
```

Comment:
Instead of `new`ing ourselves, would it make sense to accept a pre-initialized component? Thoughts being that we can support components that require arguments as a spacer?

Reply:
I considered that (but did not need it for my current use-case), so went with the simplest option, which also matches Rails spacer template implementation that does not accept any arguments.

Reply:
@BlakeWilliams do you want any other changes here? I feel like accepting components with arguments could be a future enhancement if the need arises.

Reply:
@nickcoyne up until now, ViewComponent has not handled component instantiation (and neither has Rails). I think I'd prefer we keep it that way for consistency's sake.

Reply:
@joelhawksley I've updated this to accept a component instance.